dotnet/src/Agents/Core/Chat/KernelFunctionSelectionStrategy.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents.Chat;
+
+/// <summary>
+/// Determines agent selection based on the evaluation of a <see cref="KernelFunction"/>.
+/// </summary>
+public class KernelFunctionSelectionStrategy(KernelFunction function) : SelectionStrategy
+{
+    /// <summary>
+    /// The default value for <see cref="KernelFunctionTerminationStrategy.AgentVariableName"/>.
+    /// </summary>
+    public const string DefaultAgentsVariableName = "_agents_";
+
+    /// <summary>
+    /// The default value for <see cref="KernelFunctionTerminationStrategy.HistoryVariableName"/>.
+    /// </summary>
+    public const string DefaultHistoryVariableName = "_history_";
+
+    /// <summary>
+    /// The <see cref="KernelArguments"/> key associated with the list of agent names when
+    /// invoking <see cref="KernelFunctionSelectionStrategy.Function"/>.
+    /// </summary>
+    public string AgentsVariableName { get; init; } = DefaultAgentsVariableName;
+
+    /// <summary>
+    /// The <see cref="KernelArguments"/> key associated with the chat history when
+    /// invoking <see cref="KernelFunctionSelectionStrategy.Function"/>.
+    /// </summary>
+    public string HistoryVariableName { get; init; } = DefaultHistoryVariableName;
+
+    /// <summary>
+    /// Optional arguments used when invoking <see cref="KernelFunctionSelectionStrategy.Function"/>.
+    /// </summary>
+    public KernelArguments? Arguments { get; init; }
+
+    /// <summary>
+    /// The <see cref="KernelFunction"/> invoked as selection criteria.
+    /// </summary>
+    public KernelFunction Function { get; } = function;
+
+    /// <summary>
+    /// The <see cref="Microsoft.SemanticKernel.Kernel"/> used when invoking <see cref="KernelFunctionSelectionStrategy.Function"/>.
+    /// </summary>
+    public Kernel Kernel { get; init; } = new Kernel();
```

Comment:
If this is **KernelFunction**SelectionStrategy and `KernelFunction` is required parameter, then `Kernel` should be required parameter as well, and default initialization should not be provided. Otherwise, if I pass `KernelFunctionFromPrompt` as my `KernelFunction` into this class, but use default `Kernel`, as soon as I execute my function, it will throw an exception, because it won't find any registered chat completion or text generation services in `Kernel`.

And it works in your example file because for both termination and selection strategies you use `CreateKernelWithChatCompletion` to initialize `Kernel`. If you remove it, I suspect it won't work. 

Same comment applies for `KernelFunctionTerminationStrategy`.

Reply:
That's fair.  Empty kernel doesn't make senser here.